packages/shared/src/api_types.ts

```diff
@@ -19,3 +19,8 @@ export interface UserResponse {
   login_type: "guest" | "persistent";
   username?: string;
 }
+
+export type GamesFilter = {
+  user_id: string | null;
+  variant: string | null;
```

Comment:
Nit: any reason not to make these optional?

```suggestion
  user_id?: string;
  variant?: string;
```

I think that would make it easier to represent empty/incomplete filters:

```
const filter: Ref<GamesFilter> = ref({ user_id: null, variant: "" });
```

vs.

```
const filter: Ref<GamesFilter> = ref({});
```

Reply:
Yeah that looks cleaner. The reason is that I learned there is a subtle difference of interpretation of undefined and null, that I'm thinking of when deciding between using undefined / null (e.g. for something optional).

null <-> purposefully set to "no value"
undefined <-> not set (yet)

see e.g. [https://www.geeksforgeeks.org/undefined-vs-null-in-javascript/](url)

But I don't feel strongly about this, so I'll change it as you suggested.

Reply:
made optional